typescript/widgets/src/walletIntegrations/multiProtocol.tsx

```diff
@@ -127,14 +128,55 @@ export function getAccountAddressForChain(
   if (!chainName || !accounts) return undefined;
   const protocol = multiProvider.getProtocol(chainName);
   const account = accounts[protocol];
-  if (protocol === ProtocolType.Cosmos) {
+  if (
+    protocol === ProtocolType.Cosmos ||
+    protocol === ProtocolType.CosmosNative
+  ) {
     return account?.addresses.find((a) => a.chainName === chainName)?.address;
   } else {
     // Use first because only cosmos has the notion of per-chain addresses
     return account?.addresses[0]?.address;
   }
 }
 
+export function getAddressFromAccountAndChain(
+  account?: AccountInfo,
+  chainName?: ChainName,
+) {
+  if (!account || !chainName) {
+    return 'Unknown';
+  }
+
+  // by default display the first address of the account
+  let address = account.addresses[0]?.address ?? 'Unknown';
+
+  // only in cosmos there are multiple addresses per account, in this
+  // case we display the cosmos hub address by default. If the user
+  // selects a cosmos based origin chain in the swap form that cosmos
+  // address is displayed instead
+  if (account?.protocol === ProtocolType.Cosmos) {
```

Comment:
Shouldn't tyepscript infer that account is defined now?

Reply:
Yes it should, will remove the "?" in a follow up PR